hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java

```diff
@@ -321,8 +323,10 @@ private Map<StructLike, Partition> filterNewPartitionNonExists(
               "delete and re-create with same location");
         } else {
           // this partition is need to alter, rather than delete
+          StructLikeWrapper structLikeWrapper = StructLikeWrapper.forType(table.spec().schema().asStruct());
+          structLikeWrapper.set(entry.getKey());
```

Comment:
StruckLikeWrapper is not needed.   

partitionToDelete.remove(entry.getKey) is ok